src/render/terrain.rs

```diff
@@ -559,7 +559,7 @@ impl Context {
                         // terrain data
                         wgpu::BindGroupLayoutEntry {
                             binding: 2,
-                            visibility: wgpu::ShaderStages::all(),
+                            visibility: wgpu::ShaderStages::FRAGMENT,
```

Comment:
This will prevent `Paint` method of rendering entirely. I'll add some checks though.